### PR TITLE
Integrate Message into PPA

### LINKED ISSUE #512: Integrate AMQ Message into autoreduction
Issue raised by: [developer]

### What?
We need to add the AMQ Message implementation (`message.job.Message`) from #507 into autoreduction. If this is proving large then we could try and break it up a  bit. This code is used throughout the project. Before changing any implementation code, we must check that we have tests that cover what a particular function sends to AMQ. For this issue, we should be sending the exact same information in the same format to AMQ as we were before. So in short in this issue you should:
1. Identify the areas of the code base that currently use AMQ to send / receive messages
2. Assess if this issue should be partitioned to avoid it become large (dependant on findings from (1))
3. Select an area to start in (I would suggest run_detection is probably easiest)
4. Ensure that there are tests to check what is sent to AMQ in all appropriate execution states  including the type of the data sent to AMQ (if none implement these first)
5. Change the code to use the AMQ Message class
6. Ensure the tests still pass

### Where?
* monitors/run_detection.py
* queue_processor
* autoreduction_processor
* webapp

### How?
continuation of #507 as part an initial step for the database re-design

### How to test the issue is resolved
* All test must pass
* Test must validate what is sent to AMQ
* Implementation must use AMQ Message class
* We should also validate that all is well on the development environment too

### PR DESCRIPTION
**Pull request by JEThompson97**
### Summary of work
- This PR integrates the Message class into  `post_process_admin`
- As part of the changes, the `prettify` method (which reduces the length of the reduction_script in a json dump) was replaced by a new argument of `Message.serialize`: `limit_reduction_script`.

### How to test your work
- Ensure tests pass
- Test on dev

### Additional comments
- This is part of a series of PRs to integrate the Message class throughout the database.
- Previous, un-merged PRs attempted this is various areas of the code base, but were unsuccessful due to changes that had to be made to other areas of the code (see #567)
- During this PR, I began to refactor the PPA attributes, but then reverted the change as I found it to be a larger task than anticipated

Fixes #512

**Before merging ensure the release notes have been updated**

### CODE REVIEW

**Comment by JEThompson97** [comment]
_Note to Reviewer_: I've left a `Note:` comment querying what should be done regarding `data["information"] = socket.gethostname()` - as the Message class has no 'information' attribute

File: queue_processors/autoreduction_processor/post_process_admin.py [hunk 1]
```diff
@@ -94,26 +93,17 @@ def windows_to_linux_path(path, temp_root_directory):
...
     """ Main class for the PostProcessAdmin """
 
     # pylint: disable=too-many-instance-attributes
-    def __init__(self, data, client):
-        logger.debug("json data: %s", prettify(data))
-        data["information"] = socket.gethostname()
-        self.data = data
+    def __init__(self, message, client):
+        logger.debug("Message data: %s", message.serialize(limit_reduction_script=True))
+        # Note: "information" isn't a Message attribute so is not accepted by Message.
```
Comment (ElliotAOram) [inline queue_processors/autoreduction_processor/post_process_admin.py:102]: I don't think we need to store this

**Comment by JEThompson97** [comment]
I tried to rebase with develop and for some reason git reapply all the commits of this branch (above).
Not sure why. Doesn't seem to have had any functional impact

**Comment by JackEAllen** [comment]
Performed functional tests and all works as expected.
This is good for merging 👍

**Comment by JEThompson97** [comment]
I noticed a subtle bug I introduced in #542
- Previously, manual_submission used `queue_client.serialise_data` to create a dictionary to send
- As well as package up the data given, `queue_client.serialise_data` also adds `'facility': 'ISIS'`, which I did not account for in my changes on that branch.

I've correct this bug on this branch in 8e386fe and f777c3c

**Pull request merged**